internal/transport/http_util_test.go

```diff
@@ -215,6 +218,37 @@ func (s) TestParseDialTarget(t *testing.T) {
 	}
 }
 
+type badNetworkConn struct {
+	net.Conn
+}
+
+func (c *badNetworkConn) Write([]byte) (int, error) {
+	return 0, io.EOF
+}
+
+func TestWriteBadConnection(t *testing.T) {
+	data := []byte("test_data")
+	writeBufferSize := len(data) - 1
+	writer := newBufWriter(&badNetworkConn{}, writeBufferSize, getWriteBufferPool(writeBufferSize))
+
+	syncCh := make(chan struct{})
```

Comment:
You can name this errorCh and use it to send the error from the go routine below. Then in the switch case you can check if the error received is nil 
```go
case err := <-syncCh:
  if err != nil {  \\ snipped
```

The benefit is that we don't modify a local var from a different go routing and the same channel is used for synchronization and error transmission.